### PR TITLE
IBP-4003-SelectAllPages

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Added Select All Pages Checkbox 
When select All pages clicked, query all germplasm study resource to fetch all gids 
Use _iRecordsTotal when fetching total records to display

### CODE REVIEW

File: src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasm-study-source-tab.html [hunk 1]
```diff
@@ -15,6 +15,12 @@
 					<label>Selected:</label>
 					<span class="control-label-bold">{{isAllPagesSelected ? getRecordsFiltered() : size(selectedItems)}}</span>
 				</div>
+				<div class="form-group  checkbox-inline">
+					<label style="margin-left: 15px">
+						<input type="checkbox" ng-checked="isAllPagesSelected" ng-click="onSelectAllPages()" id="check-select-all">
```
Comment (abatac) [inline src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasm-study-source-tab.html:20]: Maybe it's a good idea to hide the checkbox column if all page checkbox is selected.

File: src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasmStudySource.js [hunk 2]
```diff
@@ -358,7 +358,7 @@
 				};
 
 				$scope.getRecordsFiltered = function () {
-					return table().context[0].json && table().context[0].json['recordsFiltered'];
+					return table().context[0]._iRecordsTotal && table().context[0]._iRecordsTotal;
```
[the remark this answers is not in the document]
Comment (gelli-leafnode) [reply src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasmStudySource.js]: @abatac, 
Currently the table doesn't have the property for recordsFiltered and recordsTotal, I understand that this are set during callback. I also notice that the 3 views using this kind of approach is not fetching the recordsFiltered (subObservation, inventory (in study) and cross and selection). I used germplasm table's approach where the recordsFiltered count is stored in local variable. 
cc @darla-leafnode

[the remark this answers is not in the document]
Comment (nahuel-soldevilla) [reply src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasmStudySource.js]: @gelli-leafnode I'll look into this, it seems some of the past changes in the way we handle the ajax request may be causing this.

File: src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasmStudySource.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 (function () {
 	'use strict';
 
-	var germplasmStudySourceModule = angular.module('germplasm-study-source', []);
+	var germplasmStudySourceModule = angular.module('germplasm-study-source', [])
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasmStudySource.js:4]: Did you really mean to remove the semicolon here?

Comment (gelli-leafnode) [reply src/main/webapp/WEB-INF/static/js/trialmanager/germplasm-study-source/germplasmStudySource.js]: @darla-leafnode , 
Apologies, I think I removed it by mistake, I reverted changes

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**